media-sound/ario: bump to version 1.6

agicquel opened the pull request:
Bump EAPI to version 7.

Drop the experimental python plugin support because it does not working
anymore. The upgrade of GTK to GTK3 make it incompatible.

Remove the scrobbler USE flag because the feature was dropped.

Remove libnotify USE flag it become a dependency.

Closes: https://bugs.gentoo.org/679514
Package-Manager: Portage-2.3.51, Repoman-2.3.11

juippis commented:
You forgot the sign-off part once again, it's required before your contribution can be merged :) 
https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin

I'll make a review round, make your modifications, make a new commit and rebase and squash those updates to this original 1st commit. You can then add a sign-off when rebasing.

juippis commented on media-sound/ario/ario-1.6-r1.ebuild:
```diff
@@ -0,0 +1,55 @@
+EAPI=7
+PYTHON_COMPAT=( python2_7 )
+
+inherit autotools python-any-r1
```
Are autotools or python-any-r1 being used in here at all? Especially if you said python support was removed. If they are not, you should remove this line and the PYTHON_COMPAT above (and test rebuilding the package).

juippis commented on media-sound/ario/ario-1.6-r1.ebuild:
```diff
@@ -0,0 +1,55 @@
+
+inherit autotools python-any-r1
+
+DESCRIPTION="yayy GTK client for MPD inspired by Rhythmbox but much lighter and faster"
```
`"yayy"`? :) 
Is this suppose to mean something, or is it just a hooray? Maybe that can be taken out.

juippis commented on media-sound/ario/ario-1.6-r1.ebuild:
```diff
@@ -0,0 +1,55 @@
+	net-libs/gnutls
+	>=x11-libs/gtk+-3.21.5:3
+	dbus? ( dev-libs/dbus-glib )
+	x11-libs/libnotify
```
Unconditional deps come before conditional ones, please keep this list sorted.

juippis commented on media-sound/ario/ario-1.6-r1.ebuild:
```diff
@@ -0,0 +1,55 @@
+RDEPEND=">=dev-libs/glib-2.39.4:2
+	dev-libs/libgcrypt:0=
+	dev-libs/libunique:1
+	>=dev-libs/libxml2-2.7.4:2
```
Add a slot operator, `:2`.

a17r replied:
what slot operator, @juippis?

juippis replied:
Woops, I think I meant to say these version restrictions can be lifted since oldest ones in portage are way above the requirements... :)

My bad.

juippis commented on media-sound/ario/ario-1.6-r1.ebuild:
```diff
@@ -0,0 +1,55 @@
+	taglib? ( media-libs/taglib )
+	zeroconf? ( net-dns/avahi )"
+DEPEND="${RDEPEND}
+	dev-util/intltool
```
This and the deps below belong to BDEPEND instead of DEPEND. Please sort this list, too.

juippis commented on media-sound/ario/ario-1.6-r1.ebuild:
```diff
@@ -0,0 +1,55 @@
+
+DOCS=( AUTHORS )
+
+pkg_setup() {
```
If this package uses regular configure && make && make install, is this still needed?

agicquel commented:
Thank you for review. That's very kind to explain me those basic tasks, the learning curve is kinda hard. I'll fix this at the end of day like you mentioned. Sorry for the "aya" I was doing a test between two local repos and I didn't see this when I committed ;-)

agicquel commented:
Here you go :-) I learn a lot about the gentoo workflow, thanks again for your patience ;-)

juippis commented on media-sound/ario/ario-1.6-r1.ebuild:
```diff
@@ -0,0 +1,48 @@
+	dbus? ( dev-libs/dbus-glib )"
+DEPEND="${RDEPEND}
+	virtual/pkgconfig
+	sys-devel/gettext"
```
This is also a BDEPEND and goes in nls? ( ... )

juippis commented on media-sound/ario/ario-1.6-r1.ebuild:
```diff
@@ -0,0 +1,48 @@
+	virtual/pkgconfig
+	sys-devel/gettext"
+BDEPEND="dev-util/intltool
+		taglib? ( media-libs/taglib )
```
Put this line in RDEPEND.

juippis commented on media-sound/ario/ario-1.6-r1.ebuild:
```diff
@@ -0,0 +1,48 @@
+	sys-devel/gettext"
+BDEPEND="dev-util/intltool
+		taglib? ( media-libs/taglib )
+		zeroconf? ( net-dns/avahi )"
```
This also to RDEPEND.

agicquel commented:
I fixed my errors of dependencies and removed the unused dependencies like you mentioned.
I run ldd on ario binary and its plugins : https://gist.github.com/agicquel/996c925fd425b2e7f5029c9d4a3868e8 and only libgcrypt was used.

juippis commented:
LGTM apart for few minor things: 

You have 'nls' twice in IUSE. 

Your BDEPEND has one extra indentation in front of virtual/pkgconfig and nls?...

Still compiles and runs fine. Thanks for confirming the runtime deps. I get this when compiling: 
```
 * QA Notice: new icons were found installed but icon cache
 * has not been updated:
 *   /usr/share/icons/hicolor/96x96/apps/ario.png
 *   /usr/share/icons/hicolor/128x128/apps/ario.png
 *   /usr/share/icons/hicolor/24x24/apps/ario.png
 *   /usr/share/icons/hicolor/64x64/apps/ario.png
 *   /usr/share/icons/hicolor/scalable/apps/ario.svg
 *   /usr/share/icons/hicolor/32x32/apps/ario.png
 *   /usr/share/icons/hicolor/72x72/apps/ario.png
 *   /usr/share/icons/hicolor/48x48/apps/ario.png
 *   /usr/share/icons/hicolor/22x22/apps/ario.png
 *   /usr/share/icons/hicolor/256x256/apps/ario.png
 *   /usr/share/icons/hicolor/16x16/apps/ario.png
 * Please make sure to call xdg_icon_cache_update()
 * in pkg_postinst() and pkg_postrm() phases of appropriate pkgs.
```

so inherit xdg-utils and add pkg_postinst and pkg_postrm phases with `xdg_icon_cache_update` line in them.

You can update the sign-off tag status by editing Github issue title and appending `[please reassign]` there.

Zlogene commented on media-sound/ario/ario-1.6-r1.ebuild:
```diff
@@ -0,0 +1,45 @@
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="dbus debug +idle nls taglib zeroconf nls"
```
you stated `nls` two times

Zlogene commented on media-sound/ario/ario-1.6-r1.ebuild:
```diff
@@ -0,0 +1,45 @@
+	zeroconf? ( net-dns/avahi )"
+DEPEND="${RDEPEND}"
+BDEPEND="dev-util/intltool
+		virtual/pkgconfig
```
fix your tabs here

agicquel commented:
I have made the change requested. I hope I am finally good :-)

Zlogene commented on media-sound/ario/ario-1.6-r1.ebuild:
```diff
@@ -0,0 +1,55 @@
+
+pkg_postrm() {
+	xdg_icon_cache_update
+}
```
no new line

agicquel commented:
Yes you're right. New line added.

bircoph commented on media-sound/ario/ario-1.6-r1.ebuild:
```diff
@@ -0,0 +1,55 @@
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="dbus debug +idle nls taglib zeroconf"
```
Compared to 1.5.1-r1 libnotify, python and scrobbler USE flags are gone. Why? Does upstream support them no longer? Were any new features added?

agicquel replied:
It is explained in the commit message ;-)

bircoph commented:
Why do you commit revision 1 to the tree?
media-sound/ario/ario-1.6-r1.ebuild

This is a new version to the tree, so it should be without "-r1" suffix.

agicquel commented:
You are right about the ebuild name, I'll fix soon ;-)

mgorny commented on media-sound/ario/ario-1.6.ebuild:
```diff
@@ -0,0 +1,55 @@
+		$(use_enable taglib) \
+		$(use_enable zeroconf avahi)
+}
+
```
Could you try adding something like:

```
src_install() {
	default
	find "${D}" -name '*.la' -delete || die
}
```

in order to remove .la files, and see whether the plugins work correctly without them?

evilscientress commented:
Is this issue dead?

I tried the ebuild including the suggested patch by @mgorny and it works.

mgorny commented:
@agicquel, are you still there?

mgorny commented:
Closing due to no activity from submitter.

@evilscientress, are you interested in taking this over? If so, feel free to copy this over to your fork, and open a new PR.

evilscientress commented:
Will do so, can't say how soon, work is pilling up on my desk at the moment.
